cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,153 @@
+def test_proxy_valid(appliance, provider, request):
+    """Configures the specific valid proxy for each provider"""
+    logger.info("Begin Test Valid Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    settings_local(appliance, [provider.type], True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(OpenStackProvider))
+def test_proxy_invalid(appliance, provider, request):
+    """Configures the specific invalid proxy for each provider"""
+    logger.info("Begin Test Invalid Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    settings_local(appliance, [provider.type], False)
+    restart_appliance(appliance)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(OpenStackProvider))
+def test_proxy_remove(appliance, provider, request):
+    """Removes the specific valid proxy for all providers"""
+    logger.info("Begin Test to Remove Proxy")
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, [provider.type], provider)
+
+    appliance.ssh_client.run_command(
+        "rm -f /var/www/miq/vmdb/config/settings.local.yml")
+    restart_appliance(appliance)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(OpenStackProvider))
+def test_proxy_override(appliance, provider, request):
+    """Configures a bad default and then a valid proxy for each provider"""
+    logger.info("Begin Test Override Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, ['default', provider.type], provider)
+
+    logger.info("Test to fail first")
+    settings_local(appliance, ["default"], False)
+    restart_appliance(appliance)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+    logger.info("Then test to pass using provider specific override")
+    settings_local(appliance, ['default', provider.type], True)
+    restart_appliance(appliance)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def settings_local(appliance, proxy_type, valid=True):
+    """Creates the file string and sends it to the appliance"""
```

Comment:
**Required**
A docstring specifying proxy_type is a list and describing the use of `valid`.

An example of the yaml format you're trying to write out here.